### PR TITLE
Travis: openPMD-api Compile

### PR DESCRIPTION
**Pull request by ax3l**
Test openPMD I/O builds.
Adds no input files for runtime checks yet.

### CODE REVIEW

File: Regression/WarpX-tests.ini [hunk 1]
```diff
@@ -271,7 +271,7 @@ buildDir = .
 inputFile = Examples/Tests/Langmuir/inputs_3d_rt
 runtime_params = electrons.ux=0.01 electrons.xmax=0.e-6 warpx.fields_to_plot=Ex jx electrons.plot_vars=w ux Ex
 dim = 2
-addToCompileString =
+addToCompileString = USE_OPENPMD=FALSE
```
Comment (MaxThevenet) [inline Regression/WarpX-tests.ini]: Isn't that the default?

Comment (ax3l) [reply Regression/WarpX-tests.ini]: not in the build matrix entries I added here.

Either way, I'll wait for #723 to land and will then refactor this PR.

**Comment by MaxThevenet** [comment]
This PR seems to add 2 compilations to `WARPX_CI_REGULAR_CARTESIAN` (increasing the build time to 46 min, close to the limit): while all tests are compiled with
```sh
make -j2 AMREX_HOME=/tmp/ci-cepsfuyxib/amrex/  DEBUG=FALSE USE_ACC=FALSE USE_MPI=TRUE USE_OMP=TRUE DIM=3 USE_OPENPMD=TRUE   COMP=g++ TEST=TRUE
```
for 2D and 3D, `TwoParticle_electrostatic` and `collisionXZ` are compiled with
```sh
make -j2 AMREX_HOME=/tmp/ci-cepsfuyxib/amrex/  DEBUG=FALSE USE_ACC=FALSE USE_MPI=TRUE USE_OMP=TRUE DIM=3   COMP=g++ TEST=TRUE USE_ASSERTION=TRUE
make -j2 AMREX_HOME=/tmp/ci-cepsfuyxib/amrex/  DEBUG=FALSE USE_ACC=FALSE USE_MPI=TRUE USE_OMP=TRUE DIM=2   COMP=g++ TEST=TRUE USE_ASSERTION=TRUE
```
respectively. Is that intended?

**Comment by ax3l** [comment]
Yes, compilation times goes up since we now cover an additional large part of code during compile, although I would like to see if one can optimize it further, e.g. by re-structuring includes.

I don't fully understand the case you mention for the two setups. The only thing I try to manually disable is currently two nompi cases that are not supported yet (include/link deps) with an MPI-enabled libopenPMD-api.

**Comment by MaxThevenet** [comment]
I don't know, I just went through the TravisCI build logs and there are two more compilations than on current master branch, not sure I understand. Maybe we need to update `prepare_file_travis.py` if we want to avoid that.

**Comment by MaxThevenet** [comment]
@ax3l we can discuss this on zoom if my last comment wasn't clear. But I agree it would be nice to test that WarpX compiles with OpenPMD.

**Comment by ax3l** [comment]
@MaxThevenet what do you think, can we merge this?
It slightly increases compile time, but also because it tests more parts of the code base now.

File: .travis.yml [hunk 2]
```diff
@@ -32,9 +32,13 @@ install:
     - export CEI_CMAKE="/home/travis/.local/bin/cmake"
     - python -m pip install --upgrade git+https://github.com/yt-project/yt.git
     - sudo curl -L -o /usr/local/bin/cmake-easyinstall https://git.io/JvLxY && sudo chmod a+x /usr/local/bin/cmake-easyinstall
+    - sudo cmake-easyinstall git-https://github.com/openPMD/openPMD-api.git
+        -DopenPMD_USE_PYTHON=OFF -DBUILD_TESTING=OFF -DBUILD_EXAMPLES=OFF
+        -DCMAKE_INSTALL_PREFIX=/usr/local
```
Comment (MaxThevenet) [inline .travis.yml:37]: Could we make this conditional? It is not required for `WARPX_CI_REGULAR_CARTESIAN`, so if we can spare a few minutes there that's always good.

Comment (ax3l) [reply .travis.yml:37]: sure, this will save 2min.

Comment (MaxThevenet) [reply .travis.yml:37]: Indeed, more that 2 minutes on a job that takes >40 min, where the limit is 50 min.

Comment (ax3l) [reply .travis.yml]: fully agree, no irony implied :)

Comment (MaxThevenet) [reply .travis.yml]: Oops, sorry :disappointed:.
Thanks!

Comment (ax3l) [reply .travis.yml]: I should not have written "sure", sorry for that :)

**Pull request merged**